lockup/README.md

```diff
@@ -1,107 +1,117 @@
 # Lockup / Vesting contract
 
-# PLEASE WAIT UNTIL [ISSUE 141](https://github.com/near/core-contracts/issues/141) IS RESOLVED BEFORE DEPLOYING THE CONTRACT
-
 ## Overview
 
-This contract acts as an escrow that locks and holds an owner's tokens for a lockup period. A lockup period starts
-from the moment transfers are enabled by voting and lasts for the specified duration.
-It's also possible to lock tokens until the absolute timestamp. In this case the tokens will be available after the
-lockup period or after the timestamp, whichever is later.
+This contract acts as an escrow that locks and holds an owner's tokens for a lockup period.
+The contract consists of lockup and vesting processes that go simultaneously.
+High-level overview could be found [in Near Docs](https://docs.near.org/docs/tokens/lockup).
+
+A lockup period starts from the specified timestamp (or, if not specified, from the moment when [transfers were enabled by voting](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/) on 2020-10-13, 18:38:58 UTC or `1602614338293769340` nanoseconds unix time) and lasts for the specified duration.
+Tokens will be unlocked linearly.
+
+Vesting is an additional mechanism. It also locks the tokens, and it allows to configure 2 more options:
+1. Ability to terminate tokens vesting and refund non-vested tokens back.
+2. Cliff vesting period.
 
-If transfers are not enabled yet, the contract keeps the account ID of the transfer poll contract.
-When the transfer poll is resolved, it returns the timestamp when it was resolved and it's used as the beginning of the
-lockup period.
+The owner can add a full access key to the account if all conditions are met:
+- Transfers are enabled;
```

Comment:
This way it is much easier to comprehend :+1: 

I suggest providing more context whenever possible, so here we can provide the links to the initial voting plan and clarify that is already done:

```suggestion
- [Transfers are enabled](https://near.org/blog/near-mainnet-is-now-community-operated/) ([Phase II launch on MainNet enabled transfers](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/));
```